server/plugin.go

```diff
@@ -55,6 +71,12 @@ type Plugin struct {
 	// NPS survey.
 	userSurveyMaxDelay time.Duration
 
+	// blockSegmentEvents prevents the plugin from sending events to Segment during testing.
+	blockSegmentEvents bool
+
+	// now provides access to time.Now in a way that is mockable for unit testing.
+	now func() time.Time
```

Comment:
Is this the only way to do this? It seams hacky. Maybe convert it to an interface? Or use https://github.com/bouk/monkey?

Reply:
If you come a with a way to do this better, I would be very interested.